Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableDpad.java

```diff
@@ -8,43 +8,144 @@
 
 import android.content.res.Resources;
 import android.graphics.Bitmap;
+import android.graphics.Canvas;
 import android.graphics.Rect;
 import android.graphics.drawable.BitmapDrawable;
+import android.support.annotation.IntDef;
 import android.view.MotionEvent;
-import android.view.View;
+
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
 
 /**
  * Custom {@link BitmapDrawable} that is capable
  * of storing it's own ID.
  */
-public final class InputOverlayDrawableDpad extends BitmapDrawable
+public final class InputOverlayDrawableDpad
 {
 	// The ID identifying what type of button this Drawable represents.
 	private int[] mButtonType = new int[4];
 	private int mPreviousTouchX, mPreviousTouchY;
 	private int mControlPositionX, mControlPositionY;
+	private int mWidth;
+	private int mHeight;
+	private BitmapDrawable mDefaultStateBitmap;
+	private BitmapDrawable mPressedOneDirectionStateBitmap;
+	private BitmapDrawable mPressedTwoDirectionsStateBitmap;
+	@PressState
+	private int mPressState = STATE_DEFAULT;
+
+	@Retention(SOURCE)
```

Comment:
The way I see it, there won't ever be a scenario where we will try to pass an integer that isn't a press state, so all it does is make the code more complicated. If you want to keep it in though, I won't stop you.